txtorcon/torconfig.py

```diff
@@ -1224,11 +1224,28 @@ def _do_setup(self, data):
         # can't just return in @inlineCallbacks-decorated methods
         defer.returnValue(self)
 
+    def _add_hidden_service(self, directory, ports, auth, ver, group_read, hs, directories):
+        if directory is not None:
```

Comment:
Another reason to prefer a bare helper method over a member-function is that "self" isn't needed at all.